src/coreclr/src/inc/corpriv.h

```diff
@@ -240,6 +240,28 @@ typedef enum CorElementTypeZapSig
 
 } CorElementTypeZapSig;
 
+//////////////////////////////////////////////////////////////////////////////
+// enum CorElementTypeMulticoreJitSig defines some additional internal ELEMENT_TYPE's
+// values that are only used by MulticoreJitSig signatures.
+//////////////////////////////////////////////////////////////////////////////
+typedef enum CorElementTypeMulticoreJitSig
+{
+    // MulticoreJitSig encoding for native value types in IL stubs. IL stub signatures may contain
+    // ELEMENT_TYPE_INTERNAL followed by ParamTypeDesc with ELEMENT_TYPE_VALUETYPE element
+    // type. It acts like a modifier to the underlying structure making it look like its
+    // unmanaged view (size determined by unmanaged layout, blittable, no GC pointers).
+    //
+    // ELEMENT_TYPE_NATIVE_VALUETYPE_MULTICOREJITSIG is used when encoding such types to MulticoreJit.
+    // The signature looks like this: ET_NATIVE_VALUETYPE_MULTICOREJITSIG ET_VALUETYPE <token>.
+    // See code:MulticoreJitSig.GetSignatureForTypeHandle and code:SigPointer.GetTypeHandleThrowing
+    // where the encoding/decoding takes place.
+    ELEMENT_TYPE_NATIVE_VALUETYPE_MULTICOREJITSIG = 0x4b,
```

Comment:
I think reusing the constants defined in CorElementTypeZapSig for the multicore JIT encoding should be fine.

My concern about using ngen-specific functionality for this multi-core JIT recording was just about the preferred zap module algorithm, reusing the rest is ok.

Reply:
Thank you. I will update it.